plugins/utils/which-key.nix

```
@@ -559,26 +559,72 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin config {
       '';
       visible = false;
     };
+
+    mappingsWithOpts = mkOption {
+      type =
+        with types;
+        listOf (submodule {
+          options = {
+            spec = mkOption {
+              type = listOf (attrsOf anything);
+              description = ''
+                WhichKey mappings following the same setup as `plugins.which-key.settings.spec`.
+              '';
+            };
+            opts = mkOption {
+              type = attrsOf anything;
+              description = ''
+                WhichKey mapping options to apply to all the mapping in corresponding `plugins.which-key.mappingsWithOpts`.
```

I think the description should explain how this interacts with the `settings` we pass to `setup`; I assume it merges using vim's table extend function, so we could say can be used to **override** global options defined in `settings`?

e.g.
```suggestion
                Used to add or override WhichKey settings, only for these mappings.
                Follows the same format as `plugins.which-key.settings`.
```

I may have misinterpreted what the opts are for though?

---

Upstream's .add method does a table.insert https://github.com/folke/which-key.nvim/blob/6c1584eb76b55629702716995cca4ae2798a9cca/lua/which-key/init.lua#L49

---

I assume the queue is just to allow for `setup` and `add` to be used in any order?

---

These aren't overriding a global option though, I thought it was basically just applying mapping options to a list instead of having to declare them in each mapping. (That was my understanding)

---

My current understanding is that isn't the case. Rather, the "nested" structure is still supported in the spec (`add`s first argument) and a second `opts` argument can also be supplied to set _other_ (non-spec) options at the same time. AFAICT this is undocumented.

Does this make the PR redundant?

---

> My current understanding is that isn't the case. Rather, the "nested" structure is still supported in the spec (`add`s first argument) and a second `opts` argument can also be supplied to set _other_ (non-spec) options at the same time. AFAICT this is undocumented.
> 
> Does this make the PR redundant?

If that is the case, I would agree it is redundant. I would basically just show different examples and update documentation in this instead...